optimism/src/mips/interpreter.rs

```diff
@@ -198,11 +207,24 @@ pub fn interpret_rtype<Env: InterpreterEnv>(env: &mut Env, instr: RTypeInstructi
 
 pub fn interpret_jtype<Env: InterpreterEnv>(env: &mut Env, instr: JTypeInstruction) {
     match instr {
-        JTypeInstruction::Jump => (),
+        JTypeInstruction::Jump => {
+            // > The address stored in a j instruction is 26 bits of the address
+            // > associated with the specified label. The 26 bits are achieved by
+            // > dropping the high-order 4 bits of the address and the low-order 2
+            // > bits (which would always be 00, since addresses are always
+            // > divisible by 4).
+            // Source: https://max.cs.kzoo.edu/cs230/Resources/MIPS/MachineXL/InstructionFormats.html
+            let addr = (env.get_instruction_part(InstructionPart::RS) << 21)
+                + (env.get_instruction_part(InstructionPart::RT) << 16)
+                + (env.get_instruction_part(InstructionPart::RD) << 11)
+                + (env.get_instruction_part(InstructionPart::Shamt) << 6)
+                + (env.get_instruction_part(InstructionPart::Funct));
+            env.set_instruction_pointer(addr * 4);
+        }
         JTypeInstruction::JumpAndLink => (),
     };
     // TODO: Don't halt.
-    env.set_halted(Env::constant(1));
+    // env.set_halted(Env::constant(1));
```

Comment:
We still want to halt while jal is unimplemented. Add a return to the block instead.